File: javascript/1380.Lucky-Numbers-in-a-Matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[]}
 */
var luckyNumbers = function (matrix) {
  const rows = matrix.length;
  const cols = matrix[0].length;
  const lucky = [];

  const rows_min = [];
  const cols_max = [];

  for (let r = 0; r < rows; r++) {
    rows_min.push(Math.min(...matrix[r]));
  }

  for (let c = 0; c < cols; c++) {
    let max = -Infinity;

    for (let r = 0; r < rows; r++) {
      max = Math.max(max, matrix[r][c]);
    }

    cols_max.push(max);
  }

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (matrix[r][c] === rows_min[r] && matrix[r][c] === cols_max[c]) {
        lucky.push(matrix[r][c]);
      }
    }
  }

  return lucky;
};

File: python/1380.Lucky-Numbers-in-a-Matrix.py
class Solution:
    def luckyNumbers(self, matrix: List[List[int]]) -> List[int]:
        lucky = []
        rows = len(matrix)
        cols = len(matrix[0])
        rows_min = []
        cols_max = []

        for r in range(rows):
            rows_min.append(min(matrix[r]))

        for c in range(cols):
            max_val = float('-inf')
            for r in range(rows):
                max_val = max(max_val, matrix[r][c])
            cols_max.append(max_val)

        for r in range(rows):
            for c in range(cols):
                if matrix[r][c] == rows_min[r] and matrix[r][c] == cols_max[c]:
                    lucky.append(matrix[r][c])

        return lucky